lib/dry/auto_inject/strategies/kwargs.rb

```diff
@@ -44,24 +48,27 @@ def initialize(#{initialize_params})
         end
 
         def define_initialize_with_splat(super_method)
-          super_kwarg_names = super_method.parameters.select { |type, _| [:key, :keyreq].include?(type) }.map(&:last)
-          super_kw_params = super_kwarg_names.map { |name| "#{name}: kwargs[:#{name}]" }.join(', ')
-
-          # Pass through any non-dependency args if the super method accepts `**args`
-          if super_method.parameters.any? { |type, _| type == :keyrest }
-            if super_kw_params.empty?
-              super_kw_params = '**kwargs'
-            else
-              super_kw_params += ', **kwargs'
-            end
-          end
+          super_kwarg_names = super_method.parameters.each_with_object([]) { |(type, name), names|
+            names << name if [:key, :keyreq].include?(type)
+          }
 
           instance_mod.class_eval <<-RUBY, __FILE__, __LINE__ + 1
             def initialize(*args, **kwargs)
-              super(*args, #{super_kw_params})
+              dependency_names = [#{dependency_map.names.map { |name| ":#{name}" }.join(", ")}]
```

Comment:
This now does much more on every `initialize`, could we instead assemble a static method body in a way it worked before?

Reply:
@flash-gordon Does https://github.com/dry-rb/dry-auto_inject/pull/41/commits/d042dd7707d38bfc4c1ae145880b75901a481de9 seem better to you?